synapse/handlers/auth.py

```diff
@@ -2350,6 +2365,49 @@ async def get_username_for_registration(
 
         return None
 
+    async def get_displayname_for_registration(
```

Comment:
Not a huge deal, but it might make sense to make a helper function which does most of this, these could then become lightweight wrappers of something like:

```python
async def get_displayname_for_registration(self, uia_results, params):
    return await self._try_callbacks(self.get_displayname_for_registration_callbacks, uia_results, params)

async def _try_callbacks(self, callbacks, *args, **kwargs):
    ....
```

Reply:
(This could definitely be a follow-up too, just noting that it seems we have a lot of boilerplate code here.)

Reply:
Agreed, I'd definitely like to have some standardised wrapping like this as a follow-up. (see also the aside mentioned in the description of https://github.com/matrix-org/synapse/issues/11031)